src/movies.js

```diff
@@ -1,16 +1,40 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+    const directorsNames = movies.map((movie) => {
+      return movie.director
+    }); 
+      return directorsNames; 
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
-
+function howManyMovies(movies) {
+  const countMovies = movies.filter((movie) => {
+    if (movie.director === "Steven Spielberg" && movie.genre.indexOf('Drama') !== -1) {
+      return movie;
+    }    
+  });
+  return countMovies.length;
+}
+    
+ 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  const average = movies.reduce((acc, curr) => {
+    console.log(`${acc} .... ${curr}`);
+  return acc + curr;
+  });
+  const averageRounded = average/movies.length;
+  return averageRounded.toFixed(2);
```

Comment:
This is a tricky one! toFixed() method actually returns a string, so when you use it, you need to convert your result back to number.

i have mentioned before, bur very useful if you read the error you are getting from the test. Cause this might give you clues how to solve it.

Failed test says:
Function "scoresAverage"
should return a number

Expected: "number"
Received: "string"

One option to convert string to a number:
Number(averageRounded.toFixed(2));